api/android/api/src/main/java/org/nnsuite/nnstreamer/SingleShot.java

```diff
@@ -83,30 +84,52 @@ public SingleShot(@NonNull File model, @Nullable TensorsInfo in, @Nullable Tenso
 
     /**
      * Invokes the model with the given input data.
-     * Even if the model has flexible input data dimensions,
-     * input data frames of an instance of a model should share the same dimension.
+     * If the model has flexible input data dimensions, input information for this
+     * run of the model can be passed. This changes the currently set input information
+     * for this instance of the model. The corresponding output information can be
+     * extracted.
      *
      * Note that this has a default timeout of 3 seconds.
      * If an application wants to change the time to wait for an output,
      * set the timeout using {@link #setTimeout(int)}.
      *
      * @param in The input data to be inferred (a single frame, tensor/tensors)
+     * @param inInfo The input tensors information
      *
      * @return The output data (a single frame, tensor/tensors)
      *
      * @throws IllegalStateException if failed to invoke the model
      * @throws IllegalArgumentException if given param is null
      */
-    public TensorsData invoke(@NonNull TensorsData in) {
+    public TensorsData invoke(@NonNull TensorsData in, @Nullable TensorsInfo inInfo) {
```

Comment:
@kparichay please update the instrumentation tests (APITestSingleShot.java) - invoke() with changed params

Reply:
updated.